main.go

```diff
@@ -94,51 +97,102 @@ func mainError() error {
 		}
 	}
 
-	azureConfigValidatorConfig := azureupdate.AzureConfigValidatorConfig{
-		K8sClient: k8sClient,
-		Logger:    newLogger,
-	}
-	azureConfigValidator, err := azureupdate.NewAzureConfigValidator(azureConfigValidatorConfig)
-	if err != nil {
-		return microerror.Mask(err)
+	var azureConfigValidator *azureupdate.AzureConfigValidator
+	{
+		azureConfigValidatorConfig := azureupdate.AzureConfigValidatorConfig{
+			K8sClient: k8sClient,
```

Comment:
We've been moving towards using the controller-runtime client everywhere. Wherever we can use it, I'd say let's use it. The different `Config` objects would depend on client.Client and you would just need to pass `k8sclient.CtrlClient()` instead of `k8sclient`.

Reply:
but how do I get a G8sClient out of a CtrlClient?

Reply:
You don't need `G8sClient` anymore. CtrlClient works by looking into the object that is given to it (which it will then fill w/ data).

e.g. AzureConfigs:
```go
azureConfig := &providerv1alpha1.AzureConfig{}
err := ctrlClient.Get(ctx, client.ObjectKey{Name: "mycluster", Namespace: "default"}, azureConfig)
if err != nil {
    panic("ftw")
}
```

Reply:
ok thanks let me try

Reply:
switched to ctrlClient everywhere